src/AdminListing.php

```diff
@@ -144,9 +144,11 @@ private function init() {
     public function processRequestAndGet(Request $request, array $columns = ['*'], $searchIn = null, callable $modifyQuery = null, $locale = null) {
         // process all the basic stuff
         $this->attachOrdering($request->input('orderBy', $this->model->getKeyName()), $request->input('orderDirection', 'asc'))
-            ->attachSearch($request->input('search', null), $searchIn)
-            ->attachPagination($request->input('page', 1), $request->input('per_page', $request->cookie('per_page', 10)));
+            ->attachSearch($request->input('search', null), $searchIn);
 
+        if(!$request->input('bulk')){
+            $this->attachPagination($request->input('page', 1), $request->input('per_page', $request->cookie('per_page', 10)));
+        }
```

Comment:
@RichardDominik I think this requires a comment so everyone is sure what this condition is about.

Reply:
@RichardDominik I would also change `return $this->get($columns);` because we don't need to return nothing more than primary key when  bulk

Reply:
For what reason we need all ids for bulks? It is for bulk operation for multiple pages in listing?

Reply:
yes, it is used when you click "check all items" option